src/backend/gl/gl_common.c

```diff
@@ -1213,13 +1215,29 @@ bool gl_last_render_time(backend_t *base, struct timespec *ts) {
 		return false;
 	}
 
-	GLuint64 time;
-	glGetQueryObjectui64v(gd->frame_timing[gd->current_frame_timing ^ 1],
-	                      GL_QUERY_RESULT, &time);
-	ts->tv_sec = (long)(time / 1000000000);
-	ts->tv_nsec = (long)(time % 1000000000);
-	gl_check_err();
-	return true;
+	// We need obtain proc pointer for glGetQueryObjectui64v function.
+	// In OpenBSD this function no appear publicly, glXGetProcAddress resolve the issue.
+	if(ptrglGetQueryObjectui64v == NULL) {
+		ptrglGetQueryObjectui64v = (PFNGLGETQUERYOBJECTUI64VPROC)
+			glXGetProcAddress("glGetQueryObjectui64v");
+
+		GLuint64 time;
+		ptrglGetQueryObjectui64v(gd->frame_timing[gd->current_frame_timing ^ 1],
+					 GL_QUERY_RESULT, &time);
+		ts->tv_sec = (long)(time / 1000000000);
+		ts->tv_nsec = (long)(time % 1000000000);
+		gl_check_err();
+	}
+	else {
+		GLuint64 time;
+		ptrglGetQueryObjectui64v(gd->frame_timing[gd->current_frame_timing ^ 1],
+					 GL_QUERY_RESULT, &time);
+		ts->tv_sec = (long)(time / 1000000000);
+		ts->tv_nsec = (long)(time % 1000000000);
+		gl_check_err();
+	}
+
+ 	return true;
```

Comment:
there is no need to copy-and-paste code here.

a good place for GetProcAddress is `gl_init`.